src/v/v8_engine/script_dispatcher.h

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "bytes/iobuf.h"
+#include "model/metadata.h"
+
+#include <seastar/core/condition-variable.hh>
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/shared_ptr.hh>
+
+#include <absl/container/node_hash_map.h>
+#include <absl/container/node_hash_set.h>
+#include <v8_engine/script.h>
+
+namespace v8_engine {
+
+struct script_compile_params {
+    script_compile_params(ss::sstring function_name, iobuf code)
+      : function_name(std::move(function_name))
+      , code(std::move(code)) {}
+
+    ss::sstring function_name;
+    iobuf code;
+    size_t max_heap_size_bytes{10000};
+    size_t run_timeout_ms{200};
+};
+
+template<typename Executor>
+class script_dispatcher {
+public:
+    explicit script_dispatcher(Executor& executor)
+      : _executor(executor) {}
+
+    // Insert only put compilation params to map.
+    bool insert(
+      const model::topic_namespace& topic,
+      ss::sstring function_name,
+      iobuf code) {
+        return _compilation_params
+          .emplace(
+            topic,
+            script_compile_params(std::move(function_name), std::move(code)))
+          .second;
+    }
+
+    bool remove(const model::topic_namespace& topic) {
```

Comment:
please document this function. for example, it's really not clear from looking at the code what exactly the return value means.